BEAMS3D/Sources/beams3d_init.f90

```diff
@@ -57,790 +58,788 @@ SUBROUTINE beams3d_init
-            CALL EZspline_init(NI_spl_s(i),k,bcs1_s,ier)
-            IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init9b',ier)
-            NI_spl_s(i)%x1          = NI_AUX_S(1:k)
-            NI_spl_s(i)%isHermite   = 0
-            CALL EZspline_setup(NI_spl_s(i),NI_AUX_F(i,1:k),ier,EXACT_DIM=.true.)
-            IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init10b',ier)
-            IF (lverb .and. ANY(NI_AUX_F(i,:)>0)) WRITE(6,'(A,I1,A,F9.5,A,F9.5,A,I3,A,I2)') '   Ni(',i,')= [', &
-                        MINVAL(NI_AUX_F(i,1:k))*1E-20,',',MAXVAL(NI_AUX_F(i,1:k))*1E-20,'] E20 m^-3;  M: ',&
-                        NINT(NI_AUX_M(i)/1.66053906660E-27),' amu;  Z: ',NI_AUX_Z(i)
-         END DO
-         ! ZEFF
-         IF (nzeff>0) THEN
-            CALL EZspline_init(ZEFF_spl_s,nzeff,bcs1_s,ier)
-            IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init7',ier)
-            ZEFF_spl_s%isHermite   = 0
-            ZEFF_spl_s%x1          = ZEFF_AUX_S(1:nzeff)
-            CALL EZspline_setup(ZEFF_spl_s,ZEFF_AUX_F(1:nzeff),ier,EXACT_DIM=.true.)
-            IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init8',ier)
-            IF (lverb) WRITE(6,'(A,F9.5,A,F9.5,A,I4)') '   Zeff = [', &
-                        MINVAL(ZEFF_AUX_F(1:nzeff)),',',MAXVAL(ZEFF_AUX_F(1:nzeff)),'];  NZEFF: ',nzeff
-         END IF
-         ! POTENTIAL
-         IF (npot>0) THEN
-            CALL EZspline_init(POT_spl_s,npot,bcs1_s,ier)
-            IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init9',ier)
-            POT_spl_s%x1          = POT_AUX_S(1:npot)
-            POT_spl_s%isHermite   = 0
-            CALL EZspline_setup(POT_spl_s,POT_AUX_F(1:npot),ier,EXACT_DIM=.true.)
-            IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init10',ier)
-            IF (lverb) WRITE(6,'(A,F9.5,A,F9.5,A,I4)') '   V    = [', &
-                        MINVAL(POT_AUX_F(1:npot))*1E-3,',',MAXVAL(POT_AUX_F(1:npot))*1E-3,'] kV;  NPOT: ',npot
-         END IF
-
-         IF (lverb) THEN
-            WRITE(6,'(A,F9.5,A)') '   PLASMA_MASS =  ',plasma_mass/1.66053906660E-27,' amu' 
-            WRITE(6,'(A,F9.5,A)') '   PLASMA_ZMEAN =  ',plasma_zmean,' [Z]' 
-         END IF
-         
+      ! NION
+      DO i = 1, NION
+         k = COUNT(NI_AUX_S .ge. 0)
+         CALL EZspline_init(NI_spl_s(i),k,bcs1_s,ier)
+         IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init9b',ier)
+         NI_spl_s(i)%isHermite   = 0
+         NI_spl_s(i)%x1          = NI_AUX_S(1:k)
+         CALL EZspline_setup(NI_spl_s(i),NI_AUX_F(i,1:k),ier,EXACT_DIM=.true.)
+         IF (ier /=0) CALL handle_err(EZSPLINE_ERR,'beams3d_init10b',ier)
+         IF (lverb .and. ANY(NI_AUX_F(i,:)>0)) WRITE(6,'(A,I1,A,F9.5,A,F9.5,A,I3,A,I2,A,F8.2)') '   Ni(',i,')= [', &
+            MINVAL(NI_AUX_F(i,1:k))*1E-20,',',MAXVAL(NI_AUX_F(i,1:k))*1E-20,'] E20 m^-3;  M: ',&
+            NINT(NI_AUX_M(i)/1.66053906660E-27),' amu;  Z: ',NI_AUX_Z(i), ';  S_MAX_NI = ',s_max_zeff
```

Comment:
Shouldn't this be s_max_ni?

Reply:
s_max_zeff is handled the same way as nzeff, which in our setup logic is used for NI_AUX_*, so  I didn't create an independent s_max_ni.